hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,621 @@
+            if 'state' in init_values and record.state == 'confirm':
+                return 'hr_timesheet_sheet.mt_timesheet_confirmed'
+            elif 'state' in init_values and record.state == 'done':
+                return 'hr_timesheet_sheet.mt_timesheet_approved'
+        return super(Sheet, self)._track_subtype(init_values)
+
+    @api.model
+    def _needaction_domain_get(self):
+        empids = self.env['hr.employee'].search(
+            [('parent_id.user_id', '=', self.env.uid)])
+        if not empids:
+            return False
+        return ['&', ('state', '=', 'confirm'),
+                ('employee_id', 'in', empids.ids)]
+
+
+class SheetAccount(models.Model):
+    _name = "hr_timesheet.sheet.account"
+    _description = "Timesheet Sheets by Period"
+    _auto = False
+    _order = 'name'
+
+    name = fields.Many2one(
+        comodel_name='account.analytic.account',
+        string='Project / Analytic Account',
+        readonly=True,
+    )
+    sheet_id = fields.Many2one(
+        comodel_name='hr_timesheet.sheet',
+        string='Sheet',
+        readonly=True,
+    )
+    total = fields.Float(
+        string='Total Time',
+        digits=(16, 2),
+        readonly=True,
+    )
+
+    _depends = {
+        'account.analytic.line': ['account_id', 'date', 'unit_amount',
+                                  'user_id'],
+        'hr_timesheet.sheet': ['date_start', 'date_end', 'user_id'],
+    }
+
+    @api.model_cr
+    def init(self):
+        drop_view_if_exists(self._cr, 'hr_timesheet_sheet_account')
+        self._cr.execute("""create view hr_timesheet_sheet_account as (
+            select
+                min(l.id) as id,
+                l.account_id as name,
+                s.id as sheet_id,
+                sum(l.unit_amount) as total
+            from
+                account_analytic_line l
+                    LEFT JOIN hr_timesheet_sheet s
+                        ON (s.date_end >= l.date
+                            AND s.date_start <= l.date
+                            AND s.user_id = l.user_id)
+            group by l.account_id, s.id
+        )""")
+
+
+class SheetLine(models.TransientModel):
+    _name = 'hr_timesheet.sheet.line'
+    _description = 'Timesheet Sheet Line'
+
+    sheet_id = fields.Many2one(
+        comodel_name='hr_timesheet.sheet',
+        ondelete='cascade',
+    )
+    date = fields.Date(
+        string='Date',
+    )
+    project_id = fields.Many2one(
+        comodel_name='project.project',
+        string='Project',
+    )
+    task_id = fields.Many2one(
+        comodel_name='project.task',
+        string='Task',
+    )
+    value_x = fields.Char(
+        string='Date Name',
+    )
+    value_y = fields.Char(
+        string='Project Name',
+    )
+    unit_amount = fields.Float(
+        string="Quantity",
+        default=0.0,
+    )
+    count_timesheets = fields.Integer(
+        default=0,
+    )
+
+    @api.onchange('unit_amount')
+    def onchange_unit_amount(self):
```

Comment:
aren't we doing too many things here?
In any case, please, add a docstring saying what we do.

Reply:
Seems "too many things" but are "just a few". It's just that are many conditions. I will add docstring explaining it well.

Reply:
Added docstring. I will be sure to fully add full tests for this method.

Reply:
Method fully tested.